### ISSUE TITLE
ttnn::moe needs to apply the expert mask

### ISSUE THREAD

**Issue by iwroszTT** [tenstorrent/tt-metal#43734]
### Is your feature request related to a problem? Please describe

ttnn::moe ignores the expert mask that is provided by the caller. In consequence, current models and tests pre-apply the mask before calling into ttnn::moe. Application of the expert mask should be fused into ttnn::moe

See for example _models/tt_transformers/tt/mixtral_moe.py_
```
gate_logits_1SB8 = ttnn.matmul(input_i_1SBH, self.gates_H8, ...)        # [1, S, B, 64]
gate_logits_1SB8 = ttnn.add(gate_logits_1SB8, self.top8_mask_11B_64)    # ← pre-applies expert mask

if mode == Mode.DECODE:
    weights_1SB1 = ttnn.moe(gate_logits_1SB8, self.top8_mask_11B_64, self.top2_mask_11BB, 32)
```

### Describe the solution you'd like

ttnn::moe should implement the reference computation:
```
val, ind = torch.topk(input_tensor + expert_mask_tensor, k)
return torch.sum(torch.softmax(val + topk_mask_tensor, dim=-1) * (ind == 0), dim=-1)
```
That is including the sum: `input_tensor + expert_mask_tensor`

All uses of ttnn::moe in tt-metal need to be updated, so that the mask is applied only once.
Update the doc string if necessary.

### Additional context

Look into this PR, which solved an issue partially caused by the missing support of expert mask: https://github.com/tenstorrent/tt-metal/pull/43264

**Comment by iwroszTT** [comment]
Should wait until [issue](https://github.com/tenstorrent/tt-metal/issues/39795) is solved
